dev/django.Dockerfile

```diff
@@ -17,13 +33,28 @@ COPY ./pyproject.toml /opt/django-project/pyproject.toml
 COPY ./setup.cfg /opt/django-project/setup.cfg
 COPY ./setup.py /opt/django-project/setup.py
 
-# Copy git folder for setuptools_scm
-COPY ./.git/ /opt/django-project/.git/
+# Copy the pre-commit config so that the pre-commit environments can be
+# constructed later.
+COPY ./.pre-commit-config.yaml /opt/django-project/.pre-commit-config.yaml
+
+# Copy git folder for setuptools_scm. Make it owned by the user so that the
+# pre-commit prep steps later don't complain about "dubious ownership" of the
+# git repository.
+COPY --chown=${USERID}:${GROUPID} ./.git/ /opt/django-project/.git/
 
 # Don't install as editable, so that when the directory is mounted over with `docker compose`, the
 # installation still exists (otherwise the dandiapi.egg-info/ directory would be overwritten, and
 # the installation would no longer exist)
 RUN pip install /opt/django-project[dev]
 
+# Switch to the normal user.
+USER $LOGIN
+
 # Use a directory name which will never be an import name, as isort considers this as first-party.
 WORKDIR /opt/django-project
+
+# Run the pre-commit hooks (without --all-files, so that it won't actually run
+# on any files) to create a reusable package cache.
+RUN pre-commit run ruff-fix-only --hook-stage manual
+RUN pre-commit run ruff-format-check --hook-stage manual
+RUN pre-commit run codespell
```

Comment:
I think you can just do

```suggestion
# Run the pre-commit hooks (without --all-files, so that it won't actually run
# on any files) to create a reusable package cache.
RUN pre-commit
```
here, right? Then, we don't need to remember to update the Dockerfile when adding new pre-commit hooks.

Reply:
Very cool!